### PR TITLE
docs(ja): translate `Architecture` into Japanese

### PR DESCRIPTION
**Pull request by seo1nk**
## Summary

Partial work of https://github.com/biomejs/biome/issues/880 .
I've translated `internals/architecture.mdx`.

### CODE REVIEW

File: website/src/content/docs/ja/internals/architecture.mdx [hunk 1]
```diff
@@ -0,0 +1,160 @@
+---
+title: Architecture
+description: Biome の内部動作
+---
+
+このドキュメントでは Biome の内部機能の一部と、それらがプロジェクト内でどのように利用されているかについて説明します。
```
Comment (TaKO8Ki) [inline website/src/content/docs/ja/internals/architecture.mdx:6]:
```suggestion
このドキュメントでは、 Biome の内部機能の一部と、それらがプロジェクト内でどのように利用されているかについて説明します。
```

File: website/src/content/docs/ja/internals/architecture.mdx [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+- 頑健性: 言語の構文エラーに遭遇した後でも解析を再開できるパーサー
+- 回復可能性: エラーが発生した箇所を理解でき、正しい情報を生成することで解析を再開できるパーサー
+
+パーサーの回復可能性は科学的なルールに基づかず、厳格なルールは存在しません。
+これは、解析対象やエラー発生箇所によって予測された方法で、パーサーが自己回復できる可能性があることを示します。
+
+パーサーは誤った構文を処理することから利用者を保護するため、 `Bogus` ノードも使用します。
+このノードは、構文エラーに起因する破損コードを装飾するのに利用されます。
+
+以下の例では `while` の括弧が抜けていますが、パーサーは上手く回復し、適切な CST でコードを表現できています。
```
Comment (TaKO8Ki) [inline website/src/content/docs/ja/internals/architecture.mdx:62]:
```suggestion
以下の例では、 `while` の括弧が抜けていますが、パーサーは上手く回復し、適切な CST でコードを表現できています。
```

File: website/src/content/docs/ja/internals/architecture.mdx [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+
+CST を構築するためには、パーサーはエラーに対して頑健かつ回復可能である必要があります:
+- 頑健性: 言語の構文エラーに遭遇した後でも解析を再開できるパーサー
+- 回復可能性: エラーが発生した箇所を理解でき、正しい情報を生成することで解析を再開できるパーサー
+
+パーサーの回復可能性は科学的なルールに基づかず、厳格なルールは存在しません。
+これは、解析対象やエラー発生箇所によって予測された方法で、パーサーが自己回復できる可能性があることを示します。
+
+パーサーは誤った構文を処理することから利用者を保護するため、 `Bogus` ノードも使用します。
+このノードは、構文エラーに起因する破損コードを装飾するのに利用されます。
```
Comment (TaKO8Ki) [inline website/src/content/docs/ja/internals/architecture.mdx:60]:
```suggestion
パーサーは誤った構文を処理することからコンシューマーを保護するため、 `Bogus` ノードも使用します。
このノードは、構文エラーに起因する破損したコードを装飾するのに利用されます。
```

**Review by TaKO8Ki — CHANGES_REQUESTED** [review]
Thank you for the contribution. I left some suggestions.

**Comment by seo1nk** [comment]
@TaKO8Ki 
Thank you, fix completed!

File: website/src/content/docs/ja/internals/architecture.mdx [hunk 4]
```diff
@@ -0,0 +1,160 @@
+---
+title: Architecture
```
Comment (TaKO8Ki) [inline website/src/content/docs/ja/internals/architecture.mdx:2]:
```suggestion
title: アーキテクチャ
```

**Review by TaKO8Ki — CHANGES_REQUESTED** [review]
I left one more suggestion. After addressing it, LGTM :)

**Review by nissy-dev — APPROVED** [review]
Thanks!

**Review by TaKO8Ki — APPROVED** [review]
Thanks!

**Pull request merged**